Dockerfile.ubuntu

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:20.04
+
+#Timezone settings
+ENV TZ=Europe/Uzhgorod
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
+#Install dependencies
+RUN apt update \
+ && apt install -y python3-pip tor openssl \
+ gcc libffi-dev musl-dev make \
+ pkg-config
+
+#Setup Tor
+RUN echo "ControlPort 9051" >> /etc/tor/torrc \
+ && echo "CookieAuthentication 1" >> /etc/tor/torrc
+
+# Set up a non-root user
+RUN useradd --uid 1000 --shell /bin/bash --create-home znuser
+WORKDIR /home/znuser
+
+#Add Zeronet source
+COPY . .
+VOLUME /home/znuser/data
+RUN chown znuser:znuser -R data
+
+USER znuser
+ENV PATH="/home/znuser/.local/bin:${PATH}"
+
+RUN pip3 install virtualenv \
+ && virtualenv zeroenv \
+ && . zeroenv/bin/activate \
+ && pip3 install -r requirements.txt
+
+RUN . zeroenv/bin/activate \
+ && python3 -V \
+ && python3 -m pip list \
+ && tor --version \
+ && openssl version
```

Comment:
. zeroenv/bin/activate \ 

you don't need this, I see it 3 different times in the dockerfile 
also python3 version and other versions don't need... Not important because you already in a virtual environment. you use pip3 and venv which all python3 higher than 3.4.

Reply:
Do you understand what is virtualenv and how dockerfile run action acts? Each run is like separate shell, you cannot activate that in first run and proceed calling binaries from activated env in second run. 
Why venv and not virtualenv? I'm actually not sure why do you force venv or virtualenv that adds new virtual layer in seperate docker container that only runs docker and tor?
Versions here is only for verbose info in logs.